app/scripts/modules/amazon/src/loadBalancer/TargetGroup.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import autoBindMethods from 'class-autobind-decorator';
+import { $timeout } from 'ngimport';
+import { get, isEqual, last, orderBy } from 'lodash';
+import { PathNode } from 'angular-ui-router';
+import { Subscription } from 'rxjs';
+
+import { HealthCounts, LoadBalancerInstances, LoadBalancerServerGroup, ReactInjector } from '@spinnaker/core';
+
+import { IAmazonApplicationLoadBalancer, ITargetGroup } from 'amazon/domain/IAmazonLoadBalancer';
+
+import './targetGroup.less';
+
+export interface ITargetGroupProps {
+  loadBalancer: IAmazonApplicationLoadBalancer;
+  targetGroup: ITargetGroup;
+  showServerGroups: boolean;
+  showInstances: boolean;
+}
+
+export interface ITargetGroupState {
+  active: boolean;
+}
+
+@autoBindMethods
+export class TargetGroup extends React.Component<ITargetGroupProps, ITargetGroupState> {
+  private stateChangeListener: Subscription;
+  private baseRef: string;
+
+  constructor(props: ITargetGroupProps) {
+    super(props);
+    this.state = { active: this.isActive() };
+
+    const { stateEvents } = ReactInjector;
+
+    this.stateChangeListener = stateEvents.stateChangeSuccess.subscribe(
+      () => {
+        const active = this.isActive();
+        if (this.state.active !== active) {
+          this.setState({active});
+        }
+      }
+    );
+  }
+
+  private isActive(): boolean {
+    const { targetGroup } = this.props;
+    return ReactInjector.$state.includes('**.targetGroupDetails', {region: targetGroup.region, accountId: targetGroup.account, name: targetGroup.name, vpcId: targetGroup.vpcId, provider: targetGroup.cloudProvider});
+  }
+
+  private loadDetails(event: React.MouseEvent<HTMLElement>): void {
+    event.persist();
+    const { $state } = ReactInjector;
+    $timeout(() => {
+      const { loadBalancer, targetGroup } = this.props;
+      // anything handled by ui-sref or actual links should be ignored
+      if (event.defaultPrevented || (event.nativeEvent && event.nativeEvent.defaultPrevented)) {
+        return;
+      }
+      event.stopPropagation();
+      const params = {
+        loadBalancerName: loadBalancer.name,
+        region: targetGroup.region,
+        accountId: targetGroup.account,
+        name: targetGroup.name,
+        vpcId: targetGroup.vpcId,
+        provider: targetGroup.cloudProvider,
+      };
+
+      if (isEqual($state.params, params)) {
+        // already there
```

Comment:
note: if the state params are already equal, the $state.go() will be effectively ignored